Just getting the basics there

Flodder23 opened the pull request:
Made the interface and implemented a pretty simple Conway's Game of Life simulator

Commit:
1.0

Simulates Conway's Game of Life
Has a pause/unpause feature

Commit:
Create README.md

Commit:
1.1

Added Option to go forward by one turn using the forward arrow

Commit:
Just moved the README file

max-kov commented on Game of Game of Life.py:
```diff
@@ -0,0 +1,110 @@
+
+import pygame as pg, math as m, time as t, random as r
+
+Width=50    #C 50 How many squares wide the board is
```
every constant here should be removed and added to config file #1

joegibby replied:
done

max-kov commented on Game of Game of Life.py:
```diff
@@ -0,0 +1,110 @@
+            Next=1
+    return Next
+
+def Reset():
```
Global variables are bad, do `Board = CleanBoard()` instead, where `CleanBoard()` creates an empty board

joegibby replied:
Done

max-kov commented on Game of Game of Life.py:
```diff
@@ -0,0 +1,110 @@
+Size=20     #C 20 The size of the sides of each square (in pixels)
+Cushion=5   #C 10 How far the board extends beyond the visible amount
+Edge=m.ceil(Size/7)    #C Size/15 The gap between each cell
+Paused=True #Does this need a description?
```
No, it doesn't. Instead of making a big comment it's better to make a good variable name.

joegibby replied:
I just made the comments to be sure; they are mostly pretty self-explainitory and it just insures I don't accidently get mixed up. Also, smaller variable names are easier to deal with (eg. I put "Edge" instead of "GapBetweenEachCell" to make it easier to type

max-kov commented on Game of Game of Life.py:
```diff
@@ -0,0 +1,110 @@
+LEFT CLICK to make a Board "alive".\n
+RIGHT CLICK to Kill a cell.\n
+Press SPACE to pause/unpause the game.\n
+When paused, click RIGHT ARROW to go forward one turn.""")
```
Nobody is going to see this, a better way of doing this would be adding a menu or putting the instructions in the readme.

joegibby replied:
I have put them in the readme

joegibby replied:
Done

max-kov commented on Game of Game of Life.py:
```diff
@@ -0,0 +1,110 @@
+
+while True:
+    for event in pg.event.get(): #Checks for SPACE input for pausing/unpausing
+        x,y=pg.mouse.get_pos()
```
Make pygame event check into a function

joegibby replied:
Done, it's called "CheckUserInput()"

max-kov commented on Game of Game of Life.py:
```diff
@@ -0,0 +1,110 @@
+                else:
+                    Board[a][b].Birth(Fate)
+    for a in range(Cushion,Width+Cushion):     #This is seperate to when the values
+        for b in range(Cushion,Height+Cushion):# are changed because not all
```
function

max-kov commented on Game of Game of Life.py:
```diff
@@ -0,0 +1,110 @@
+            Draw(Board[a][b].NextState,a,b,colour) #Draws the cell as desired
+    for a in range(0,Width+(2*Cushion)):
+        for b in range(0,Height+(2*Cushion)):
+            Board[a][b].CurrentState=Board[a][b].NextState #Updates CurrentState variable
```
I'd make 2 boards - current board and future board, that way you won't need this loop, and it will be easier to understand what's going on. Using more variables is ok if they make the program easier to read.

joegibby replied:
I thought about that but that doesn't let me create a class, which is something I am trying to include as without it it will be difficult to get a good mark

Commit:
Updated parts as Max suggested

see "Done" comments in Pull Request

Commit:
Added a config file

max-kov commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+
+__pycache__/config.cpython-36.pyc
```
change it to `__pychache__`, because right now it is only ignoring one file, not the whole folder.

joegibby replied:
done

max-kov commented on Game of Game of Life.py:
```diff
@@ -1,18 +1,7 @@
 ### a,b refers to cell (a,b) whereas x,y refers to pixel coordinates ###
 
 import pygame as pg, math as m, time as t, random as r
```
Change to one import per line and don't do `import as`. That is used rarely and only for popular packages like numpy, and if you use it on every package the code is harder to read because you need to always keep in mind that the package was renamed. Read https://www.python.org/dev/peps/pep-0008/#imports for more

joegibby replied:
done

max-kov commented on Game of Game of Life.py:
```diff
@@ -1,18 +1,7 @@
-Background=(120,120,120)    #C (120,120,120) The colour of the background
-Dead=0
-Square=1
+from config import *
```
Change to `import config` and use `config.constant` for aforementioned reasons.

joegibby replied:
done

Commit:
Oops

just corrected a height/width mix-up

max-kov commented on Game of Game of Life.py:
```diff
@@ -81,7 +70,9 @@ def CheckUserInput():
 Screen.fill(Background)
 pg.display.set_caption("Game of Life")
-Board=CleanBoard()
+Board=CleanBoard()#A 2d matrix representing the board
```
Not really even a mistake, but there is no need for extra comments on self-explanatory stuff. If I see `Board=CleanBoard()` I know that this is the game board and it is empty. If I want to know the variable type of the board I go and look at the function. Read this https://www.python.org/dev/peps/pep-0008/#comments

joegibby replied:
done

max-kov commented on Game of Game of Life.py:
```diff
@@ -0,0 +1,98 @@
+
+def CheckUserInput():
+    global Paused
+    global OneTurn
```
Remove globals and make this function return 2 variables. 
```
def func():
...
return Var1, Var2

...

Var1, Var2 = func()
```

joegibby replied:
done

max-kov commented on Game of Game of Life.py:
```diff
@@ -0,0 +1,98 @@
+                if Fate==0:
+                    Board[a][b].Kill()
+                else:
+                    Board[a][b].Birth(Fate)
```
I'd make everything in this loop functions so it would look like
```
while True:
CheckInput()
CheckBoard()
NextTurn()
```

max-kov commented on Game of Game of Life.py:
```diff
@@ -0,0 +1,98 @@
+    total[Board[a+1][b+1].CurrentState]+=1
+    total[Board[a-1][b-1].CurrentState]+=1
+    total[Board[a+1][b-1].CurrentState]+=1
+    total[Board[a-1][b+1].CurrentState]+=1
```
Make this into a loop

joegibby replied:
This will get in the way later on, when I (hopefully) introduce the new cell types - some of them don't take into account every square eg. only diagonals.

Commit:
Update .gitignore

Commit:
Added some of Max's suggested changes

Namely:
-improved the imports
-removed some superfluous comments
-removed global variables
-added functions to make the main loop easier to read
-Changed 0 and 1 values to commit.Dead and commit.Square respectively to make it clearer which meaning is significant in this use

Commit:
PEP8

Updated code to fit with PEP8 specification

joegibby commented:
Stop spamming @max-kov
